### PR TITLE
[InstallAppleCertificateV2] fixing the cert common name extraction regex

### PR DESCRIPTION
**Pull request by ajaykn**
Issue posted here:
https://developercommunity.visualstudio.com/content/problem/967683/installapplecertificate-vsts-task-signingidentity.html

We have a certificate with SLASH in its commonName.
when tested with a self signed cert:

```
subject=/C=KR/ST=Seoul/L=Seoul/O=MS/OU=MS/CN=test_cert_/_with_slash/emailAddress=test@ms.com
```
And checked the below variables, its returning `test_cert_`

$(APPLE_CERTIFICATE_SIGNING_IDENTITY) 
$(InstallAppleCertificate.signingIdentity)

When verified the code, Regex thats being used in extracting the commonName is not handling these cases.

https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/Common/ios-signing-common/ios-signing-common.ts#L552

The current regex `/\/CN=([^/]+)/` looks for just slash.
The fix is to have a positive lookahead for /O= or /OU= or /C= or /emailAddress= etc.

so came up with this regex`/\/CN=.*?(?=\/[A-Za-z]+=|$)/`

Let me know any concerns.

### CODE REVIEW

**Comment by ajaykn** [comment]
@hashtagchris , @leantk 
can you have a look into this please.

**Comment by giovannidegani** [comment]
Hi. We have production builds that cannot move forward in our company in MS App Center due to this issue. Any idea when it will be merged ?

File: Tasks/Common/ios-signing-common/ios-signing-common.ts [hunk 1]
```diff
@@ -547,11 +547,13 @@ export async function getP12Properties(p12Path: string, p12Pwd: string): Promise
...
+                // Example value1: "/UID=E848ASUQZY/CN=iPhone Developer: Chris Sidi (7RZ3N927YF)/OU=DJ8T2973U7/O=Chris Sidi/C=US"
+                // Example value2: "/UID=E848ASUQZY/CN=iPhone Developer: Chris / Sidi (7RZ3N927YF)/OU=DJ8T2973U7/O=Chris Sidi/C=US"
+                // Example value3: "/UID=E848ASUQZY/OU=DJ8T2973U7/O=Chris Sidi/C=US/CN=iPhone Developer: Chris Sidi (7RZ3N927YF)"
                 // Extract the common name.
-                const matches: string[] = value.match(/\/CN=([^/]+)/);
-                if (matches && matches[1]) {
-                    commonName = matches[1].trim();
+                const matches: string[] = value.match(/\/CN=.*?(?=\/[A-Za-z]+=|$)/);
+                if (matches && matches[0]) {
+                    commonName = matches[0].replace("/CN=", "");
```
Comment (lkillgore) [inline Tasks/Common/ios-signing-common/ios-signing-common.ts:556]: Does this need a `.trim()`?

**Comment by leantk** [comment]
For the version number updates, can you actually just reset `Patch` to 0 and `Minor` to 168 please? Once that and the above issue Luke pointed out is addressed then I will approve. Sorry for the delay on this. We have been a little backed up :-(

**Comment by romeomorcia** [comment]
It would be amazing to have this bugfix released asap 👍 
Thank you guys. Following the post!

File: Tasks/Common/ios-signing-common/ios-signing-common.ts (hunk 1 above)
Comment (ajaykn) [reply Tasks/Common/ios-signing-common/ios-signing-common.ts]: yeah, added to be on safe side :)

**Comment by ajaykn** [comment]
> For the version number updates, can you actually just reset `Patch` to 0 and `Minor` to 168 please?

@leantk I have made the version fix by resetting the patch and bumping the minor version.
Not sure what's with 168 here ? because each task has it own versioning number.

**Comment by leantk** [comment]
> @leantk I have made the version fix by resetting the patch and bumping the minor version.
> Not sure what's with 168 here ? because each task has it own versioning number.

@ajaykn the Minor version we set to the sprint that we are on when we are making a change. I know that is not visible to you so I apologize for that. What you did is fine since it's incrementing it no matter what

**Comment by leantk** [comment]
Thanks for the contribution here! Merging now. The deployment of the change won't start at least till the 7th though due to a freeze we are in due to Covid-19. The official Microsoft statement can be found below but we will do our best to get this out ASAP

https://azure.microsoft.com/en-us/blog/our-commitment-to-customers-and-microsoft-cloud-services-continuity/

**Comment by giovannidegani** [comment]
What is missing for the PR to be close ? We need to see this change reach App Center for us to be able to setup new builds.

**Comment by ajaykn** [comment]
@leantk, would like to know the status on the deployment?

**Comment by leantk** [comment]
This is being deployed not and is out for a smaller subset of customers. It should continue to rollout this week and next barring no incidents

**Comment by leantk** [comment]
This change should be everywhere so I'm closing this out but please reopen if you are still seeing issues

**Comment by giovannidegani** [comment]
I think we are having some kind of regression related to this, our builds are failing and it is showing in App Center the certificate name being trimmed right where the "/" was ( iPhone Distribution: LEGO System A/S  is the CN for our enterprise cert )

**Comment by anatolybolshakov** [comment]
Hi @giovannidegani could you please create an issue for it and share some logs so we could investigate it?

**Pull request merged**